source/extensions/filters/network/dubbo_proxy/hessian_utils.cc

```diff
@@ -0,0 +1,534 @@
+    *size = delta_length + 1;
+    return result;
+
+  case 0x30:
+  case 0x31:
+  case 0x32:
+  case 0x33:
+    if (offset + 2 > buffer.length()) {
+      throw EnvoyException("buffer underflow");
+    }
+
+    delta_length = (code - 0x30) * 256 + buffer.peekInt<int8_t>(offset + 1);
+    if (delta_length + 2 + offset > buffer.length()) {
+      throw EnvoyException("buffer underflow");
+    }
+
+    buffer.copyOut(offset + 2, delta_length, allocStringBuffer(&result, delta_length));
+    *size = delta_length + 2;
+    return result;
+
+  case 0x53:
+    if (offset + 3 > buffer.length()) {
+      throw EnvoyException("buffer underflow");
+    }
+
+    delta_length = buffer.peekBEInt<uint16_t>(offset + 1);
+
+    if (delta_length + 3 + offset > buffer.length()) {
+      throw EnvoyException("buffer underflow");
+    }
+
+    buffer.copyOut(offset + 3, delta_length, allocStringBuffer(&result, delta_length));
+    *size = delta_length + 3;
+    return result;
+
+  case 0x52:
+    if (offset + 3 > buffer.length()) {
+      throw EnvoyException("buffer underflow");
+    }
+
+    delta_length = buffer.peekBEInt<uint16_t>(offset + 1);
+    buffer.copyOut(offset + 3, delta_length, allocStringBuffer(&result, delta_length));
+    size_t next_size = 0;
+    result.append(peekString(buffer, &next_size, delta_length + 3 + offset));
+    *size = next_size + delta_length + 3;
+    return result;
+  }
+  throw EnvoyException(fmt::format("hessian type is not string {}", code));
+}
+
+std::string HessianUtils::readString(Buffer::Instance& buffer) {
+  size_t size;
+  std::string result(peekString(buffer, &size));
+  buffer.drain(size);
+  return result;
+}
+
+long HessianUtils::peekLong(Buffer::Instance& buffer, size_t* size, uint64_t offset) {
+  ASSERT(buffer.length() > offset);
+  long result;
+  uint8_t code = buffer.peekInt<uint8_t>(offset);
+  switch (code) {
+  case 0xd8:
+  case 0xd9:
+  case 0xda:
+  case 0xdb:
+  case 0xdc:
+  case 0xdd:
+  case 0xde:
+  case 0xdf:
+  case 0xe0:
+  case 0xe1:
+  case 0xe2:
+  case 0xe3:
+  case 0xe4:
+  case 0xe5:
+  case 0xe6:
+  case 0xe7:
+  case 0xe8:
+  case 0xe9:
+  case 0xea:
+  case 0xee:
+  case 0xef:
+
+    result = code - 0xe0;
+    *size = 1;
+    return result;
+
+  case 0xf0:
+  case 0xf1:
+  case 0xf2:
+  case 0xf3:
+  case 0xf4:
+  case 0xf5:
+  case 0xf6:
+  case 0xf7:
+  case 0xf8:
+  case 0xf9:
+  case 0xfa:
+  case 0xfb:
+  case 0xfe:
+  case 0xff:
+
+    if (offset + 2 > buffer.length()) {
+      throw EnvoyException("buffer underflow");
+    }
+
+    result = leftShift<int16_t>(code - 0xf8, 8) + buffer.peekInt<uint8_t>(offset + 1);
+    *size = 2;
+    return result;
+
+  case 0x38:
+  case 0x39:
+  case 0x3a:
+  case 0x3b:
+  case 0x3c:
+  case 0x3d:
+  case 0x3e:
+  case 0x3f:
+
+    if (offset + 3 > buffer.length()) {
+      throw EnvoyException("buffer underflow");
+    }
+
+    result = leftShift<int32_t>(code - 0x3c, 16) + (buffer.peekInt<uint8_t>(offset + 1) << 8) +
```

Comment:
peekInt<uint16_t> for the latter part.

Reply:
I am a hessian official documents to implement the code and documentation of pseudo code description is consistent, so I don't want to use `peekInt<uint16_t>` replace `(buffer.peekInt<uint8_t>(offset + 1) << 8) + buffer.peekInt<uint8_t>(offset + 2)`

http://hessian.caucho.com/doc/hessian-serialization.html#anchor17